source/dem/pw_contact_force.cc

```diff
@@ -19,5 +19,136 @@
 
 #include <dem/pw_contact_force.h>
 
+// Updates the contact information (contact_info) based on the new information
+// of particles pair in the current time step
+template <int dim>
+void
+PWContactForce<dim>::update_contact_information(
+  pw_contact_info_struct<dim> &  contact_info,
+  const ArrayView<const double> &particle_properties,
+  const double &                 dt)
+{
+  auto normal_vector = contact_info.normal_vector;
+
+  // Using velocity and angular velocity of particle as
+  // local vectors
+  Tensor<1, dim> particle_velocity;
+  particle_velocity[0] = particle_properties[DEM::PropertiesIndex::v_x];
+  particle_velocity[1] = particle_properties[DEM::PropertiesIndex::v_y];
+  if (dim == 3)
+    {
+      particle_velocity[2] = particle_properties[DEM::PropertiesIndex::v_z];
+    }
+
+  Tensor<1, dim> particle_omega;
+  particle_omega[0] = particle_properties[DEM::PropertiesIndex::omega_x];
+  particle_omega[1] = particle_properties[DEM::PropertiesIndex::omega_y];
+  if (dim == 3)
+    {
+      particle_omega[2] = particle_properties[DEM::PropertiesIndex::omega_z];
+    }
+
+  // Defining relative contact velocity
+  Tensor<1, dim> contact_relative_velocity;
+  if (dim == 3)
+    {
+      contact_relative_velocity =
+        particle_velocity +
+        cross_product_3d((((particle_properties[DEM::PropertiesIndex::dp]) /
+                           2) *
+                          particle_omega),
+                         normal_vector);
+    }
+
+  if (dim == 2)
+    {
+      contact_relative_velocity = particle_velocity;
```

Comment:
Should there not be an impact of angular velocity also in 2D? it still does alter the tangential velocity no? Maybe not something to fix now, but worth to keep in mind for future PR.